grid2op/Reward/_alertTrustScore.py

```diff
@@ -0,0 +1,136 @@
+            own risk.
+
+    Implemented as a reward to make it easier to use in the context of the L2RPN competitions, this reward is based on the "alert feature" 
+    where the agent is asked to send information about potential line overload issue on the grid after unpredictable powerline
+    disconnection (attack of the opponent).
+    The alerts are assessed once per attack and cumulated over the episode. In this scheme, this "reward" computed the assistant "score",
+    which assesses how well the agent is aware of its capacities to deal with a situation during an episode.
+    It should not be used to train an agent.
+    
+    """
+    
+    def __init__(self,
+                 logger=None,
+                 reward_min_no_blackout=-1.0,
+                 reward_min_blackout=-6.2, 
+                 reward_max_no_blackout=0.06,
+                 reward_max_blackout=27.0,
+                 reward_end_episode_bonus=9.0):
+        
+        super().__init__(logger,
+                 reward_min_no_blackout,
+                 reward_min_blackout, 
+                 reward_max_no_blackout,
+                 reward_max_blackout,
+                 reward_end_episode_bonus)
+        
+        self.reward_min = dt_float(-1.0)
+        self.reward_max = dt_float(1.0)
+        
+    def initialize(self, env):
+        self._is_simul_env = self.is_simulated_env(env)
+            
+        self.cumulated_reward = 0
+        #KPIs
+        self.total_nb_attacks = 0
+        self.total_nb_alerts = 0
+        self.alert_attack_no_blackout = 0
+        self.alert_attack_blackout = 0
+        self.no_alert_attack_no_blackout = 0
+        self.no_alert_attack_blackout = 0
+        self.blackout_encountered = False
+        return super().initialize(env)
+            
+    def reset(self, env):
+        super().reset(env)
+        self.cumulated_reward = 0
+        #KPIs
+        self.total_nb_attacks = 0
+        self.total_nb_alerts = 0
+        self.alert_attack_no_blackout = 0
+        self.alert_attack_blackout = 0
+        self.no_alert_attack_no_blackout = 0
+        self.no_alert_attack_blackout = 0
+        self.blackout_encountered = False
+            
+    def __call__(self, action, env, has_error, is_done, is_illegal, is_ambiguous):
+        self.blackout_encountered = self.is_in_blackout(has_error, is_done)
+        
+        score_ep = 0.
+        if self._is_simul_env:
+            return score_ep
+            
+        res = super().__call__(action, env, has_error, is_done, is_illegal, is_ambiguous)
+        self.cumulated_reward += res
+        
+        lines_attacked = env._time_since_last_attack == 0      
+        self.total_nb_attacks += np.sum(lines_attacked)
+        
+        lines_alerted_beforeattack = np.equal(env._time_since_last_alert, env._time_since_last_attack + 1)
+        self.total_nb_alerts += np.sum(lines_alerted_beforeattack)
```

Comment:
There is a bug here : self.total_nb_alert is cumulated at *every* step where `lines_alerted_beforeattack > 0` (not only once)

Reply:
Il faut cumuler avec la condition "lines_attacked"

np.equal(env._time_since_last_alert, env._time_since_last_attack + 1) and lines_attacked